lib/sidekiq/form526_job_status_tracker/job_tracker.rb

```diff
@@ -73,8 +73,14 @@ def with_tracking(job_title, saved_claim_id, submission_id, is_bdd = nil)
         @is_bdd = is_bdd
 
         job_try
-        yield
-        job_success
+        begin
+          yield
+        rescue
+          exception_raised = true
+          raise
+        ensure
+          job_success unless exception_raised
+        end
```

Comment:
Did you try:
```suggestion
        begin
          yield
        rescue
          raise
        else
          job_success
        end
```
https://stackoverflow.com/questions/6279956/ruby-exceptions-why-else

Reply:
Yeah, I did -- returns and breaks also skip the `else` section, which doesn't have the no-matter-what urgency of `ensure`.

```ruby
def wrapper
  begin
    puts "before yield"
    yield
  rescue
    raise
  else
    puts "after yield"
  end
end
wrapper { break }
```
This just prints "before yield".